src/components/search/SearchResults.tsx

```diff
@@ -0,0 +1,88 @@
+import Fuse from "fuse.js";
+import { useQuery } from "@tanstack/react-query";
+import { useEffect, useState } from "react";
+import PostCard from "../PostCard";
+import LoadingSpinner from "../ui/LoadingSpinner";
+
+export type ResultObj = {
+  slug: string;
+  data: {
+    title: string;
+    description: string;
+    date: string;
+  };
+};
+
+type SearchResult = {
+  item: ResultObj;
+  refIndex: number;
+  score: number;
+};
+
+const DEFAULT_NUM = 5;
+
+const SearchResults = ({ query }: { query: string }) => {
+  const [matchingResults, setMatchingResults] = useState<SearchResult[]>([]);
+  const [numOfResults, setNumOfResults] = useState(DEFAULT_NUM);
+  const { status, error, data } = useQuery<SearchResult[]>({
+    queryKey: ["search"],
+    queryFn: async () => {
+      try {
+        const res = await fetch(`/postSearch.json`);
+        if (!res.ok) throw new Error("Network response was not ok");
+        return res.json();
+      } catch (e) {
+        console.error(e);
+        return e;
+      }
+    },
+  });
+
+  const fuse = new Fuse(data, {
+    includeScore: true,
+    shouldSort: true,
+    threshold: 0.5,
+    keys: [
+      {
+        name: "data.title",
+        weight: 2,
+      },
+      {
+        name: "data.description",
+        weight: 1,
+      },
+    ],
+  });
+
+  useEffect(() => {
+    if (data) {
+      const results = fuse.search(query);
+      setMatchingResults(results);
+    }
+  }, [query, data, numOfResults]);
+
+  if (query.length === 0) return null;
```

Comment:
nit: if no results, could we show a div saying like "no results found. try searching for something else?"

(although I could be misunderstanding how this works...)

Reply:
Ahhh, I see now! Makes perfect sense 🙌